src/Fixer/PhpUnit/PhpUnitOrderedCoversFixer.php

```diff
@@ -0,0 +1,105 @@
+<?php
+
+/*
+ * This file is part of PHP CS Fixer.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *     Dariusz Rumiński <dariusz.ruminski@gmail.com>
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace PhpCsFixer\Fixer\PhpUnit;
+
+use PhpCsFixer\AbstractFixer;
+use PhpCsFixer\DocBlock\Annotation;
+use PhpCsFixer\DocBlock\DocBlock;
+use PhpCsFixer\FixerDefinition\CodeSample;
+use PhpCsFixer\FixerDefinition\FixerDefinition;
+use PhpCsFixer\Tokenizer\Token;
+use PhpCsFixer\Tokenizer\Tokens;
+
+/**
+ * @author Filippo Tessarotto <zoeslam@gmail.com>
+ */
+final class PhpUnitOrderedCoversFixer extends AbstractFixer
+{
+    /**
+     * {@inheritdoc}
+     */
+    public function getDefinition()
+    {
+        return new FixerDefinition(
+            'Order `@covers` annotation of PHPUnit tests.',
+            [
+                new CodeSample(
+'<?php
+/**
+ * @covers Foo
+ * @covers Bar
+ */
+final class MyTest extends \PHPUnit_Framework_TestCase
+{}
+'
+                ),
+            ]
+        );
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getPriority()
+    {
+        // should be run after PhpUnitFqcnAnnotationFixer
+        return -10;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function isCandidate(Tokens $tokens)
+    {
+        return $tokens->isAllTokenKindsFound([T_CLASS, T_DOC_COMMENT]);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function applyFix(\SplFileInfo $file, Tokens $tokens)
+    {
+        for ($index = $tokens->count() - 1; $index > 0; --$index) {
+            if (!$tokens[$index]->isGivenKind(T_DOC_COMMENT) || 0 === preg_match('/@covers\s.+@covers\s/s', $tokens[$index]->getContent())) {
+                continue;
+            }
+
+            $docBlock = new DocBlock($tokens[$index]->getContent());
+
+            $coversMap = [];
+            $linesToUpdate = [];
+            foreach ($docBlock->getLines() as $line) {
```

Comment:
why not use `getAnnotationsOfType` ?

Reply:
Because in this case:
```php
/**
 * @covers \Foo
 *
 *    MyComment
 * @covers \Bar
 */
```
`getAnnotationsOfType` returns also the following new lines with `MyComment` included for `\Foo`, which is contrary on how PHPUnit handles this specific annotation.

Reply:
according to phpdoc, `MyComment` is part of `@covers \Foo`.
if it would be modified with respecting of that trailing msg into:
```
 * @covers \Bar
 * @covers \Foo
 *
 *    MyComment
```

then PHPUnit would still be happy, wound't it ?

Reply:
In this case I have been more pragmatic than correct.
PHPUnit would still be happy, but the developer I think not.
Would you like to have this:
```php
/**
 * @covers \B
 * @covers \C
 * @covers \D
 * @covers \E
 * @covers \F
 * @covers \A
 *
 *  Lorem ipsum dolor sit amet, consectetur adipiscing elit,
 *  sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.
 *  Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris
 *  nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit
 *  in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint
 *  occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
 */
```
Fixed to this:
```php
/**
 * @covers \A
 *
 *  Lorem ipsum dolor sit amet, consectetur adipiscing elit,
 *  sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.
 *  Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris
 *  nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit
 *  in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint
 *  occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
 * @covers \B
 * @covers \C
 * @covers \D
 * @covers \E
 * @covers \F
 */
```
?

Reply:
we do it like that everywhere else ;)

having any text after annotation that is intended to not be part of annotation is always problematic, so yes, lets do how you showed

Reply:
Done.